lib/buffer.js

```diff
@@ -1073,77 +1234,79 @@ Buffer.prototype.writeDoubleBE = function writeDoubleBE(val, offset, noAssert) {
 
 // ES6 iterator
 
-var ITERATOR_KIND_KEYS = 1;
-var ITERATOR_KIND_ENTRIES = 3;
+if (useOldBuffer) {
+  var ITERATOR_KIND_KEYS = 1;
+  var ITERATOR_KIND_ENTRIES = 3;
 
-function BufferIteratorResult(value, done) {
-  this.value = value;
-  this.done = done;
-}
+  function BufferIteratorResult(value, done) {
+    this.value = value;
+    this.done = done;
+  }
 
-var resultCache = new Array(256);
+  var resultCache = new Array(256);
 
-for (var i = 0; i < 256; i++)
-  resultCache[i] = Object.freeze(new BufferIteratorResult(i, false));
+  for (var i = 0; i < 256; i++)
+    resultCache[i] = Object.freeze(new BufferIteratorResult(i, false));
 
-var finalResult = Object.freeze(new BufferIteratorResult(undefined, true));
+  var finalResult = Object.freeze(new BufferIteratorResult(undefined, true));
 
-function BufferIterator(buffer, kind) {
-  this._buffer = buffer;
-  this._kind = kind;
-  this._index = 0;
-}
+  function BufferIterator(buffer, kind) {
+    this._buffer = buffer;
+    this._kind = kind;
+    this._index = 0;
+  }
 
-BufferIterator.prototype.next = function() {
-  var buffer = this._buffer;
-  var kind = this._kind;
-  var index = this._index;
+  BufferIterator.prototype.next = function() {
+    var buffer = this._buffer;
+    var kind = this._kind;
+    var index = this._index;
 
-  if (index >= buffer.length)
-    return finalResult;
+    if (index >= buffer.length)
+      return finalResult;
 
-  this._index++;
+    this._index++;
 
-  if (kind === ITERATOR_KIND_ENTRIES)
-    return new BufferIteratorResult([index, buffer[index]], false);
+    if (kind === ITERATOR_KIND_ENTRIES)
+      return new BufferIteratorResult([index, buffer[index]], false);
 
-  return new BufferIteratorResult(index, false);
-};
+    return new BufferIteratorResult(index, false);
+  };
 
-function BufferValueIterator(buffer) {
-  BufferIterator.call(this, buffer, null);
-}
+  function BufferValueIterator(buffer) {
```

Comment:
Might not need any of the custom iterator implementation for the new version (i.e. can hide all this behind an if)